### PR TITLE
gh-117122: Fix pystats after incremental GC changes

### PR DESCRIPTION
**Pull request by mdboom**
This fixes the pystats build after #116206.

* Issue: gh-117122

### CODE REVIEW

File: Python/gc.c [hunk 1]
```diff
@@ -1384,7 +1384,6 @@ expand_region_transitively_reachable(PyGC_Head *container, PyGC_Head *gc, GCStat
 static void
 completed_cycle(GCState *gcstate)
 {
-    PyGC_Head *not_visited = &gcstate->old[gcstate->visited_space^1].head;
```
Comment (mdboom) [inline Python/gc.c:1387]: This is a (mostly unrelated) change to remove an unused variable.

File: Python/gc.c [hunk 2]
```diff
@@ -1421,7 +1420,7 @@ gc_collect_increment(PyThreadState *tstate, struct gc_collection_stats *stats)
         gc_set_old_space(gc, gcstate->visited_space);
         increment_size += expand_region_transitively_reachable(&increment, gc, gcstate);
     }
-    GC_STAT_ADD(1, objects_queued, region_size);
+    GC_STAT_ADD(1, objects_queued, increment_size);
```
Comment (mdboom) [inline Python/gc.c:1427]: I don't know if this is correct, but `region_size` doesn't exist.

Comment (markshannon) [reply Python/gc.c:1427]: I'm not sure if this make sense anymore. Maybe just drop the "queued" stat?

**Review by markshannon — COMMENTED** [review]
This is going to conflict with https://github.com/python/cpython/pull/117120.
Once that's in, I'll review this properly.

**Comment by markshannon** [comment]
Actually, don't worry about the merge conflict. I can fix it in my PR

**Comment by mdboom** [comment]
> I'm not sure if this make sense anymore. Maybe just drop the "queued" stat?

I leave that to you, but this stat is new as of #116206 (it's just that it never compiled).

**Pull request closed without merging**